ambry-messageformat/src/main/java/com.github.ambry.messageformat/CompositeBlobInfo.java

```diff
@@ -18,49 +18,58 @@
 import java.util.List;
 
 
-public class MultiPartMetadata {
+/**
+ * This class holds information about a composite blob parsed from the metadata blob. It contains the chunk size,
+ * total composite blob size, and a list of keys for the blob's data chunks.
+ */
+public class CompositeBlobInfo {
   public static final int UNDEFINED_CHUNK_SIZE = -1;
   public static final int UNDEFINED_TOTAL_SIZE = -1;
   private final int chunkSize;
   private final long totalSize;
   private final List<StoreKey> keys;
 
   /**
-   * Construct a {@link MultiPartMetadata} object with undefined max chunk size and total size.
+   * Construct a {@link CompositeBlobInfo} object with undefined chunk size and total size. This constructor can be used
+   * when chunk size and total size are not defined in the metadata format, such as with metadata content format V1.
    * @param keys The list of keys for this object's data chunks.
    */
-  public MultiPartMetadata(List<StoreKey> keys) {
+  public CompositeBlobInfo(List<StoreKey> keys) {
     this(UNDEFINED_CHUNK_SIZE, UNDEFINED_TOTAL_SIZE, keys);
   }
 
   /**
-   * Construct a {@link MultiPartMetadata} object.
-   * @param chunkSize The intermediate chunk size for the object in bytes.
-   * @param totalSize The total size of the object.
+   * Construct a {@link CompositeBlobInfo} object.
+   * @param chunkSize The size of each data chunk except the last, which could possibly be smaller.
+   * @param totalSize The total size of the composite blob.
    * @param keys The list of keys for this object's data chunks.
    */
-  public MultiPartMetadata(int chunkSize, long totalSize, List<StoreKey> keys) {
+  public CompositeBlobInfo(int chunkSize, long totalSize, List<StoreKey> keys) {
```

Comment:
We already use "blobInfo" for some other purpose. So, I feel like composite blob info is kind of confusing.